Cython/Build/IpythonMagic.py

```diff
@@ -106,6 +108,43 @@ def encode_fs(name):
         return name
 
 
+def get_encoding_candidates():
+    candidates = [sys.getdefaultencoding()]
+    try:
+        candidates.extend([sys.stdout.encoding, sys.stdin.encoding])
+    except AttributeError:
+        pass
+    try:
+        candidates.extend([sys.__stdout__.encoding, sys.__stdin__.encoding])
+    except AttributeError:
+        pass
+    unique_candidates = OrderedDict.fromkeys(candidates)
+    # encoding might be None (i.e. somebody redirects stdout):
+    unique_candidates.pop(None, None)
+    return list(unique_candidates)
+
+
+def prepare_captured(captured):
+    captured_bytes = captured and captured().strip()
```

Comment:
`captured` can be None in the `finally` clause in `runtests.py`, but not here.
```suggestion
    captured_bytes = captured.strip()
```

Reply:
simplified: https://github.com/cython/cython/pull/3872/commits/3efc7b11c9efd5e435146bd766f0b74d230bcb06